llvm/lib/Target/X86/X86TargetMachine.cpp

```diff
@@ -505,6 +505,9 @@ bool X86PassConfig::addRegBankSelect() {
 
 bool X86PassConfig::addGlobalInstructionSelect() {
   addPass(new InstructionSelect(getOptLevel()));
+  // Add GlobalBaseReg in case there is no SelectionDAG passes afterwards
+  if (isGlobalISelAbortEnabled())
```

Comment:
This shouldn't need to consider which selector is enabled, the passes have to check if the selection failed or not

Reply:
I don't know what is better actually. The problem is that this pass is added by SelectionDAG
```cpp
bool X86PassConfig::addInstSelector() {
  // Install an instruction selector.
  addPass(createX86ISelDag(getX86TargetMachine(), getOptLevel()));
  // For ELF, cleanup any local-dynamic TLS accesses.
  if (TM->getTargetTriple().isOSBinFormatELF() &&
      getOptLevel() != CodeGenOptLevel::None)
    addPass(createCleanupLocalDynamicTLSPass());
  addPass(createX86GlobalBaseRegPass());
  addPass(createX86ArgumentStackSlotPass());
  return false;
}
```
I see three options:
1) Add the same pass twice, however one is enough
2) Add only once depending on which selector is the last (this was implemented)
3) Extract this pass from selectors in general pipeline (I haven't tried), but it looks like this pass fixes instruction selection and should go close to it.

Reply:
I read this backwards, but I guess this is just mirroring how the fallback pipeline is added. It might make sense to have a separate post-isel insert point

Reply:
Yes, indeed, it mirrors the check from here
https://github.com/llvm/llvm-project/blob/82b95d0089bea67d1a06fa75d2a4b5e087563d6f/llvm/include/llvm/Passes/CodeGenPassBuilder.h#L836-L838
I think it would be better to create a new insert point in a separate PR?